tests/returntypes/test_new_return_types.py

```diff
@@ -77,15 +86,19 @@ def circuit(x):
         qnode = qml.QNode(circuit, dev)
         qnode.construct([0.5], {})
 
-        res = qml.execute_new(tapes=[qnode.tape], device=dev, gradient_fn=None)
+        if dev.shots is not None:
+            with pytest.warns(UserWarning, match="with finite shots; the returned"):
+                res = qml.execute_new(tapes=[qnode.tape], device=dev, gradient_fn=None)
+        else:
+            res = qml.execute_new(tapes=[qnode.tape], device=dev, gradient_fn=None)
 
         assert res[0].shape == (2**d_wires, 2**d_wires)
         assert isinstance(res[0], np.ndarray)
 
     @pytest.mark.parametrize("device", devices)
     def test_expval(self, device, shots):
         """Return a single expval."""
-        dev = qml.device(device, wires=2)
+        dev = qml.device(device, wires=2, shots=shots)
```

Comment:
Not completely in scope, but stumbled upon these cases that should be parametrized by `shots` too.

Reply:
Thanks!